### ISSUE TITLE
Errors while Configuring Tamagui

### ISSUE THREAD

**Issue by aliansyahFirdaus** [tamagui/tamagui#1556]
### Current Behavior

I'm currently facing an issue while attempting to configure Tamagui within the Bit.dev platform. When trying to integrate my Tamagui project with Bit.dev, I encounter TypeScript type inference problems that are causing errors. Specifically, I'm receiving messages such as: **"The inferred type of 'appConfig' cannot be named without a reference to..."**. This prevents me from proceeding with the integration

![Screenshot 2023-08-21 at 10 38 27](https://github.com/tamagui/tamagui/assets/74974463/cd392a46-10ae-4cb8-bd95-537af5adc91c)

**package manager using PNPM**

### Expected Behavior

I was expecting the integration of Tamagui within Bit.dev to proceed smoothly without TypeScript type inference errors. I anticipate that the configuration should allow me to seamlessly share and manage Tamagui components within the Bit.dev ecosystem.

### Tamagui Version

```markdown
"@tamagui/config": "^1.52.9",
"@tamagui/shorthands": "^1.52.9",
"@tamagui/themes": "^1.52.9",
"tamagui": "^1.52.9"
```

### Reproduction

```markdown

1. bit install tamagui @tamagui/config @tamagui/shorthands @tamagui/themes (Just like installing packages using pnpm)
2. create config

--------------------------------------------------------------------
import { config } from '@tamagui/config';
import { createTamagui } from 'tamagui'; // or '@tamagui/core'

/**
Error Message: 
The inferred type of 'appConfig' cannot be named without a reference to '.pnpm/registry.npmjs.org+@tamagui+themes@1.52.9_react@18.2.0/node_modules/@tamagui/themes/types/generated-new'. This is likely not portable. A type annotation is necessary.ts(2742)
*/
const appConfig = createTamagui(config) //Error

export type AppConfig = typeof appConfig;

declare module 'tamagui' {
  // or '@tamagui/core'

  // overrides TamaguiCustomConfig so your custom types

  // work everywhere you import `tamagui`

  interface TamaguiCustomConfig extends AppConfig {}
}
export default appConfig;
--------------------------------------------------------------------
```

### System Info

```markdown
System:
    OS: macOS 13.4.1
    CPU: (8) arm64 Apple M1
    Memory: 44.81 MB / 8.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.16.0 - ~/.nvm/versions/node/v18.16.0/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.5.1 - ~/.nvm/versions/node/v18.16.0/bin/npm
    pnpm: 7.13.6 - /usr/local/bin/pnpm
    Watchman: 2023.08.14.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 116.0.5845.96
    Safari: 16.5.2
```

**Comment by deggertsen** [comment]
Experiencing similar issues on 1.57.4.

**Comment by natew** [comment]
Fix going out in latest release `1.58.7`, not sure why ts compiler emits like this just had to override it manually.
